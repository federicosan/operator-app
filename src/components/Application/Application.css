#app > div {
  display: flex;
}

#app .panel {
  border-right: 1px solid #efefef;
}

#app .panel:last-child {
  border: 0;
  box-shadow: inset 10px 0 50px -15px #efefef;
}

#app .bar {
  height: 50px;
  width: 100%;
}

#app .panel > .top.container .bar {
  border-bottom: 1px solid #e0e0e0;
}