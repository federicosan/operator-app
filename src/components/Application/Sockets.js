export default function initSockets (socket) {
  // Successful connection
  function onSocketConnected () {
    console.log('DEBUG', 'Socket connected');
  }

  // Disconnected
  function onSocketDisconnected () {
    console.warn('DEBUG', 'Socket disconnected');
  }

  // Successful re-connected
  function onSocketReconnected () {
    console.log('DEBUG', 'Socket reconnected');
  }

  // Attempting to re-connect
  function onSocketReconnecting () {
    console.log('DEBUG', 'Socket reconnecting ...');
  }

  // Failed to re-connect after manager.reconnectionAttempts tried
  function onSocketReconnectionFailed () {
    console.error('DEBUG', 'Socket failed reconnection');
  }

  // Timeout either connecting or re-connecting
  function onSocketTimeout () {
    console.warn('DEBUG', 'Socket timeout');
  }

  // Error when connecting or re-connecting
  function onSocketConnectionError () {
    console.error('DEBUG', 'Socket connection error');
  }

  socket.on('connect', onSocketConnected);
  socket.on('connect_error', onSocketConnectionError);
  socket.on('connect_timeout', onSocketTimeout);
  socket.on('disconnect', onSocketDisconnected);
  socket.on('reconnect', onSocketReconnected);
  socket.on('reconnecting', onSocketReconnecting);
  // socket.on('reconnect_error', socketConnectionError);
  socket.on('reconnect_failed', onSocketReconnectionFailed);
  socket.on('reconnect_timeout', onSocketTimeout);
}
