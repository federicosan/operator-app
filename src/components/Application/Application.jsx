import React from 'react';
import { connect } from 'react-redux';

import OperatorPanel from '../OperatorPanel/OperatorPanel.jsx';
import ClientsPanel from '../ClientsPanel/ClientsPanel.jsx';
import MessagePanel from '../MessagePanel/MessagePanel.jsx';

import './Application.css';

export const ApplicationComponent = () => (
  <div>
    <OperatorPanel />
    <ClientsPanel />
    <MessagePanel />
  </div>
);

const mapStateToProps = state => ({

});

const mapDispatchToProps = dispatch => ({

});

const Application = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ApplicationComponent);

export default Application;
