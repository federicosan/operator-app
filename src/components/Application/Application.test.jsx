import React from 'react';

import { shallow } from 'enzyme';

import Application from './Application.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({
    ui: { settingsOpen: false },
  })),
};

describe('Chat', () => {
  it('matches snapshot', () => {
    const component = shallow(<Application store={store} />);

    expect(component).toMatchSnapshot();
  });
});
