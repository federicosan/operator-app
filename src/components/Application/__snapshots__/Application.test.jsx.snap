exports[`Chat matches snapshot 1`] = `
<Component
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
